File: app/src/main/java/me/kirimin/mitsumine/model/EntryInfo.java
package me.kirimin.mitsumine.model;

import java.util.List;

public class EntryInfo {

    private final String title;
    private final int bookmarkCount;
    private final String url;
    private final String thumbnailUrl;
    private final List<Bookmark> bookmarkList;

    public EntryInfo(String title, int bookmarkCount, String url, String thumbnailUrl, List<Bookmark> bookmarkList) {
        this.title = title;
        this.bookmarkCount = bookmarkCount;
        this.url = url;
        this.thumbnailUrl = thumbnailUrl;
        this.bookmarkList = bookmarkList;
    }

    public EntryInfo() {
        this.title = "empty";
        this.bookmarkCount = 0;
        this.url = "";
        this.thumbnailUrl = "";
        this.bookmarkList = null;
    }

    public boolean isNullObject() {
        return getTitle().equals("empty");
    }

    public String getTitle() {
        return title;
    }

    public int getBookmarkCount() {
        return bookmarkCount;
    }

    public String getUrl() {
        return url;
    }

    public String getThumbnailUrl() {
        return thumbnailUrl;
    }

    public List<Bookmark> getBookmarkList() {
        return bookmarkList;
    }
}

File: app/src/main/java/me/kirimin/mitsumine/model/EntryInfo.kt
package me.kirimin.mitsumine.model

import java.util.ArrayList

public class EntryInfo {

    public val title: String
    public val bookmarkCount: Int
    public val url: String
    public val thumbnailUrl: String
    public val bookmarkList: List<Bookmark>

    public constructor(title: String, bookmarkCount: Int, url: String, thumbnailUrl: String, bookmarkList: List<Bookmark>) {
        this.title = title
        this.bookmarkCount = bookmarkCount
        this.url = url
        this.thumbnailUrl = thumbnailUrl
        this.bookmarkList = bookmarkList
    }

    public constructor() {
        this.title = "empty"
        this.bookmarkCount = 0
        this.url = ""
        this.thumbnailUrl = ""
        this.bookmarkList = ArrayList()
    }

    public fun isNullObject(): Boolean {
        return title == "empty"
    }
}
